auto_install.sh

```diff
@@ -58,33 +62,30 @@ if [ ! -f "$CONFIG_JSON_FILE" ]; then
     exit 1
 fi
 
-OPTIONS=s:h
+if [ $# -lt 1 ] || [ $1 == '-h' ]; then
+    usage
+    exit 1
+fi
+
+XMOS_DEVICE=$1
+shift 1
+
+OPTIONS=s:gh
 while getopts "$OPTIONS" opt ; do
     case $opt in
         s )
             DEVICE_SERIAL_NUMBER="$OPTARG"
             ;;
+        g )
+            GPIO_KEY_WORD_DETECTOR_FLAG="-g"
+            ;;
         h )
             usage
             exit 1
             ;;
     esac
 done
 
-shift $(( OPTIND - 1 ))
-
-if [[ $# -ge 1 ]]; then
-  XMOS_DEVICE=$1
-else
-  echo No device specified.
```

Comment:
I don't mind the removal of the functionality to prompt the user if they fail to specify a device.  However, I would like to see the list of recognised devices printed if the user specifies a device that does not match any in the list.

Reply:
That functionality has not been removed. These are the printouts if the wrong device is used:

```
./auto_install.sh xvf1234
error: xvf1234 is not a valid device type.

usage: auto_install.sh <DEVICE-TYPE> [OPTIONS]

A JSON config file, config.json, must be present in the current working
directory. The config.json can be downloaded from developer portal and must
contain the following:
   "clientId": "<Auth client ID>"
   "productId": "<your product name for device>"

The DEVICE-TYPE is the XMOS device to setup: xvf3100, xvf3500, xvf3510, xvf3600-slave, xvf3600-master, xvf3610 or xvf3615

Optional parameters:
  -s <serial-number>  If nothing is provided, the default device serial number
                      is 123456
  -g                  Flag to enable keyword detector on GPIO interrupt
  -h                  Display this help and exit
```

Reply:
I did not understand how the script worked.  Thanks for explaining.